### ISSUE TITLE
Sending binary data

### ISSUE THREAD

**Issue by ggallea** [mostafa/xk6-kafka#159]
I am Reading data from binary file and trying to send the data I read to kafka. 
I get the following error:
cannot unmarshal object into Go struct field Message.messages.value of type []uint8

What am I doing wrong?

Thanks

**Comment by mostafa** [comment]
Hey @ggallea,

Have a look at [this example](https://github.com/mostafa/xk6-kafka/blob/main/scripts/test_bytes.js).

**Comment by ggallea** [comment]
The example is serializing a string. I have numbers (float and int32) to send
I also tried with Uint8Array and it gave me the same error

**Comment by mostafa** [comment]
@ggallea
Can you send your script and some sample data, so I can use it to debug the issue?

**Comment by ggallea** [comment]
c```
onst producer = new Writer({
    brokers: bootstrapServers,
    topic: kafkaTopic,
});

const params = open('params.bin', 'b');

export default function () {
    producer.produce({
        messages: [
            {
                key: null,
                value: params
            }
        ]
    });
}
```

**Comment by ggallea** [comment]
How can I serialize (types other then string) to bytes?

**Comment by c-meier** [comment]
Hello @mostafa,

Would it be possible to directly send an `ArrayBuffer`?

For the moment I do the following:
```js
const body = encoding.b64decode(".............")  // returns an ArrayBuffer
writer.produce({ messages: [{
    value: schemaRegistry.serialize({
        data: Array.from(new Uint8Array(body)),
        schemaType: SCHEMA_TYPE_BYTES,
    }),
    key: ...
}]})
```

It works but it means a lot of transformations.

**Comment by mostafa** [comment]
Hey @c-meier,

you can read the reasoning behind making the serdes explicit in the following PR and its linked issues:
- https://github.com/mostafa/xk6-kafka/pull/149

**Comment by c-meier** [comment]
Hello @mostafa,

My point wasn't about exporting the serdes functionality to JS (I think it works quite well). But that I think it would make more sense for `SCHEMA_TYPE_BYTES` to accept data as an [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) (or maybe [`Uint8Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array)) instead of an `Array[number]`

**Comment by mostafa** [comment]
@c-meier
Fair enough! Then I need to test it to see if and how it works. In the meantime, if you can help contribute some code, it'll be much appreciated.
